apps/condo/domains/analytics/schema/ExternalReportService.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Generated by `createservice analytics.ExternalReportService`
+ */
+
+const { GQLCustomSchema } = require('@condo/keystone/schema')
+const access = require('@condo/domains/analytics/access/ExternalReportService')
+const { ExternalReport } = require('@condo/domains/analytics/utils/serverSchema/index')
+const { EXTERNAL_REPORT_EXPIRATION_SECONDS } = require('@condo/domains/analytics/constants/constants')
+const conf = require('@condo/config')
+const get = require('lodash/get')
+const jwt = require('jsonwebtoken')
+
+const getMetabaseUrl = (externalReport, organizationId) => {
+    const METABASE_CONFIG = conf['METABASE_CONFIG'] && JSON.parse(conf['METABASE_CONFIG'])
+
+    const metabaseServerUrl = get(METABASE_CONFIG, 'url')
+    const metabaseSecret = get(METABASE_CONFIG, 'secret')
+    const dashboardId = get(externalReport, ['meta', 'dashboard'])
+
+    if (!metabaseServerUrl || !metabaseSecret || !dashboardId) {
+        return null
+    }
+
+    const payload = {
+        resource: { dashboard: dashboardId },
+        params: {
+            'organizationid': organizationId,
+        },
+        exp: Math.round(Date.now() / 1000) + EXTERNAL_REPORT_EXPIRATION_SECONDS,
+    }
+    const token = jwt.sign(payload, metabaseSecret, {}, null)
+
+    return metabaseServerUrl + '/embed/dashboard/' + token + '#bordered=false&titled=false'
+}
+
+const ExternalReportService = new GQLCustomSchema('ExternalReportService', {
+    types: [
+        {
+            access: true,
+            type: 'input ExternalReportServiceInput { dv: Int!, sender: SenderFieldInput!, id: ID!, organizationId: ID! }',
+        },
+        {
+            access: true,
+            type: 'type ExternalReportServiceOutput { title: String!, iframeUrl: String }',
+        },
+    ],
+
+    queries: [
+        {
+            access: access.canReadExternalReportService,
+            schema: 'externalReportService (data: ExternalReportServiceInput!): ExternalReportServiceOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
```

Comment:
We also can create a virtual field with arguments: https://v5.keystonejs.com/keystonejs/fields/src/types/virtual/#field-arguments